apache/superset
# fix(explore): Filters with custom SQL disappearing

### SUMMARY
When user added more than 1 filter with custom SQL and saved the chart, all filters with custom SQL except for the first one disappeared. This PR fixes that issue.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

https://user-images.githubusercontent.com/15073128/185192610-40a9c076-7540-4818-a6b2-b89d1b20c84f.mov

After:

https://user-images.githubusercontent.com/15073128/185192803-44626ed7-e1c6-42e6-ab09-41a7b464c44a.mov

### TESTING INSTRUCTIONS
1. Open explore and add a few filters with custom sql
2. Save chart
3. All added filters should still be there

1. Open a mixed chart
2. Add adhoc filters for both queries
3. Save chart
4. Make sure that adhoc filters for both queries are still there

### ADDITIONAL INFORMATION
